lib/micro/case.rb

```diff
@@ -37,9 +37,7 @@ def self.call(options = {})
     end
 
     def self.__new__(result, arg)
-      instance = new(arg)
-      instance.__set_result__(result)
-      instance
+      new(arg).tap { |instance| instance.__set_result__(result) }
```

Comment:
@linqueta Nice refactoring! But it adds a little overhead (please, check out the below example), and yes, I know that is a micro-optimization, but one of the project's goals is: `Be fast and optimized`. So I think that this kind of change would be unnecessary.

Following this discussion, what do you think about the refactoring to replace the usage of the [method `tap`](https://github.com/serradura/u-case/blob/master/lib/micro/case/result.rb#L44-L52) to increase the `u-case` performance?

```ruby
require 'bundler/inline'

gemfile do
  source 'https://rubygems.org'

  gem 'benchmark-ips', '~> 2.7', '>= 2.7.2'
end

require 'benchmark/ips'

Result = Struct.new(:value)

class Foo
  def self.__new__a(value, result)
    new(value).tap { |instance| instance.__set_result__(result) }
  end

  def self.__new__b(value, result)
    instance = new(value)
    instance.__set_result__(result)
    instance
  end

  def initialize(value)
    @value = value 
  end

  def __set_result__(result)
    @result = result
  end
end

result = Result.new(nil)

Benchmark.ips do |x|
  x.config(:time => 5, :warmup => 2)

  x.time = 5
  x.warmup = 2
  
  x.report('with tap') { Foo.__new__a(1, result) }
  
  x.report('without tap') { Foo.__new__b(2, result) }

  x.compare!
end
```